superduper/backends/base/data_backend.py

```diff
@@ -25,9 +27,11 @@ class BaseDataBackend(ABC):
     """
 
     tables_ignore: str | None = None
+    json_native: bool = False
 
     batched: bool = False
     id_field: str = 'id'
+    vector_impl: t.Type = NativeVector
```

Comment:
How about using a set of datatype implementations like this? That way, we can optimize the logic and make it easier to extend.

```python
datatypes_impl = {"vector": NativeVector, "json": "JsonXXX"}
```